packages/fluentui/react-northstar/src/components/Chat/ChatMessage.tsx

```diff
@@ -285,13 +371,53 @@ export const ChatMessage: ComponentWithAs<'div', ChatMessageProps> &
         className: chatMessageSlotClassNames.actionMenu,
         styles: resolvedStyles.actionMenu,
       }),
+      overrideProps: {
+        id: actionMenuId.current,
+      },
     });
 
-    if (!actionMenuElement) {
-      return actionMenuElement;
+    const content = actionMenuElement ? <Ref innerRef={actionsMenuRef}>{actionMenuElement}</Ref> : actionMenuElement;
+
+    return inlineActionMenu || !content ? content : <PortalInner>{content}</PortalInner>;
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    // taken from https://github.com/microsoft/fluentui/pull/17329
+    const focusableElements = actionsMenuRef?.current?.querySelectorAll(
+      '[tabindex="0"],[tabindex="-1"]:not([data-is-focusable="false"])',
+    );
+
+    if (!focusableElements) {
+      _.invoke(props, 'onKeyDown', e, props);
+      return;
     }
 
-    return <Ref innerRef={actionsMenuRef}>{actionMenuElement}</Ref>;
+    if (e.keyCode === keyboardKey.Enter) {
+      focusableElements[0].focus();
+      e.stopPropagation();
+      e.preventDefault();
+    }
+    if (e.keyCode === keyboardKey.Tab) {
+      const focusableElementsInsideMessage = e.currentTarget.querySelectorAll(
+        '[tabindex="-1"]:not([data-is-focusable="false"])',
+      );
+      if (e.shiftKey) {
+        const firstElement = focusableElementsInsideMessage[0];
+        if (e.target === firstElement) {
+          focusableElements[0].focus();
+          e.stopPropagation();
+          e.preventDefault();
+        }
+      } else {
+        const lastElement = focusableElementsInsideMessage[focusableElementsInsideMessage.length - 1];
+        if (e.target === lastElement) {
+          focusableElements[0].focus();
+          e.stopPropagation();
+          e.preventDefault();
+        }
+      }
```

Comment:
nit: is this cleaner ? 

```suggestion
      const isShift = !!e.shiftKey;
      const boundaryElement = isShift ? focusableElementsInsideMessage[0] : focusableElementsInsideMessage[focusableElementsInsideMessage.length - 1]
      if (e.target === boundaryElement) {
        focusableElements[0].focus();
        e.stopPropagation();
        e.preventDefault();
      }
```

Reply:
👍 will commit the suggestion after I updated e2e

Reply:
nice proposal from Ling :)
looking into it and on line 407 should be  `boundaryElement.focus()` ?
because there is already saved proper element based on "Shift" key was used or not...

Reply:
@YuanboXue-Amber can you take a look at @kolaps33 comment ?